pkg/types/aws/platform.go

```diff
@@ -129,6 +135,50 @@ type Platform struct {
 	UserProvisionedDNS dns.UserProvisionedDNS `json:"userProvisionedDNS,omitempty"`
 }
 
+// EIPAllocations contains Elastic IP (EIP) allocations for AWS resources
+// within the cluster.
+type EIPAllocations struct {
+	// ingressNetworkLoadBalancer is a list of IDs for Elastic IP (EIP) addresses that
+	// are assigned to the default AWS NLB IngressController.
+	// The following restrictions apply:
+	//
+	// eipAllocations can only be used with external scope, not internal.
+	// An EIP can be allocated to only a single IngressController.
+	// The number of EIP allocations must match the number of subnets that are used for the load balancer.
+	// Each EIP allocation must be unique.
+	// A maximum of 10 EIP allocations are permitted.
+	//
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	//
+	// +openshift:enable:FeatureGate=InstallEIPForDefaultNLBIngressController
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+	// +kubebuilder:validation:MaxItems=10
```

Comment:
@patrickdillon does the installer actually handle feature gates in the installconfig?

Additionally, your CEL should probably be removed as it can't be used.

Reply:
Yes, featuregates can be supplied in the install config and you can gate install config fields with feature gates like this:

https://github.com/openshift/installer/blob/0f78927350d769bb8142d0eb1a56a1b01660fa46/pkg/types/validation/installconfig.go#L1287-L1295

Reply:
@patrickdillon it looks like I need to add feature gate here https://github.com/openshift/installer/blob/master/pkg/types/aws/validation/featuregates.go#L15
and also I need add feature gate in openshift/api here https://github.com/openshift/api/blob/master/features/features.go#L39?
Am I right ?

Reply:
Also I was having a look at previous PRs and saw features gates mentioned in the types. https://github.com/openshift/api/commit/caf97963ed302881f01e67b791ae4efc6bcf723c#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR1742

Reply:
But any case it looks like I need open an openshift/api PR to add a feature gate in https://github.com/openshift/api/blob/master/features/features.go#L39

Reply:
@patrickdillon @sadasu @r4f4  @mtulio  can you PTAL ?

Reply:
Yes, you need to start with creating a feature gate in openshift/api and the bump the o/api version in the installer so we have access to that feature gate.